### ISSUE TITLE
sns_topic module "fails" new topic creation on empty "subscriptions" argument

### ISSUE THREAD

**Issue by srkiNZ84** [ansible/ansible-modules-extras#1889]
##### Issue Type:
- Bug Report
##### Ansible Version:

```
ansible 2.0.0.2
  config file = /Users/srdan/code/gdt-aws-ansible-infrsatructure/ansible.cfg
  configured module search path = Default w/o overrides
```
##### Ansible Configuration:

hash_behaviour = merge
##### Environment:

Running from: OS X
Managing: Ubuntu 14.04 LTS
##### Summary:

Despite the documentation specifying that the "subscriptions" argument is not required by the module, leaving it out results in the following error (after the AWS SNS object has been created for the first time):

```
An exception occurred during task execution. The full traceback is:
Traceback (most recent call last):
  File "/Users/srdan/.ansible/tmp/ansible-tmp-1458595733.98-257514012961217/sns_topic", line 2421, in <module>
    main()
  File "/Users/srdan/.ansible/tmp/ansible-tmp-1458595733.98-257514012961217/sns_topic", line 234, in main
    subscriptions]
TypeError: 'NoneType' object is not iterable

fatal: [localhost]: FAILED! => {"changed": false, "failed": true, "invocation": {"module_name": "sns_topic"}, "parsed": false}
```
##### Steps To Reproduce:

Create a new SNS Topic in AWS with the "sns_topic" module, leaving out the "subscriptions" argument

```
- name: Create an SNS topic
  sns_topic:
    aws_access_key: "{{ ec2_access_key }}"
    aws_secret_key: "{{ ec2_secret_key }}"
    region:         "{{ ec2_region }}"
    name: "newly-created-topic"
    display_name: "newly-created-topic"
    state: present
```
##### Expected Results:

The task runs, creates the SNS Topic object (without any subscriptions) and completes successfully.
##### Actual Results:

The task runs, creates the SNS Topic object (without any subscriptions) but throws the above listed exception and "fails".

```

TASK [sns : Create an SNS topic]
task path: /Users/srdan/code/ansible/roles/sns/tasks/main.yml:43
ESTABLISH LOCAL CONNECTION FOR USER: srdan
localhost EXEC ( umask 22 && mkdir -p "$( echo $HOME/.ansible/tmp/ansible-tmp-1458595733.98-257514012961217 )" && echo "$( echo $HOME/.ansible/tmp/ansible-tmp-1458595733.98-257514012961217 )" )
localhost PUT /var/folders/01/fswx1t811p3_q5bj53yjfhnw0000gn/T/tmpK3aeDY TO /Users/srdan/.ansible/tmp/ansible-tmp-1458595733.98-257514012961217/sns_topic
localhost EXEC LANG=C LC_ALL=C LC_MESSAGES=C /usr/bin/python /Users/srdan/.ansible/tmp/ansible-tmp-1458595733.98-257514012961217/sns_topic; rm -rf "/Users/srdan/.ansible/tmp/ansible-tmp-1458595733.98-257514012961217/" > /dev/null 2>&1
An exception occurred during task execution. The full traceback is:
Traceback (most recent call last):
  File "/Users/srdan/.ansible/tmp/ansible-tmp-1458595733.98-257514012961217/sns_topic", line 2421, in <module>
    main()
  File "/Users/srdan/.ansible/tmp/ansible-tmp-1458595733.98-257514012961217/sns_topic", line 234, in main
    subscriptions]
TypeError: 'NoneType' object is not iterable

fatal: [localhost]: FAILED! => {"changed": false, "failed": true, "invocation": {"module_name": "sns_topic"}, "parsed": false}

```

**Comment by robbwagoner** [comment]
I'm also seeing this issue: using `ansible-playbook 2.0.1.0` on Ubuntu 14.04.4 LTS.

### Playbook

``` yaml
---
- hosts: localhost
  gather_facts: no
  tasks:
    - name: Create SNS Topic
      sns_topic:
        name: autoscaling-staging
        purge_subscriptions: no
        state: present
        region: "{{ region }}"
        profile: "{{ profile | default(omit) }}"
```

### Actual Results

```
TASK [Create SNS Topic] ********************************************************
task path: /home/robb/git/infrastructure/ansible/sns.yml:10
ESTABLISH LOCAL CONNECTION FOR USER: robb
localhost EXEC /bin/sh -c '( umask 22 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1461073790.5-184778957863254 `" && echo "` echo $HOME/.ansible/tmp/ansible-tmp-1461073790.5-184778957863254 `" )'
localhost PUT /tmp/tmpb5KZmY TO /home/robb/.ansible/tmp/ansible-tmp-1461073790.5-184778957863254/sns_topic
localhost EXEC /bin/sh -c 'LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 /usr/bin/python /home/robb/.ansible/tmp/ansible-tmp-1461073790.5-184778957863254/sns_topic; rm -rf "/home/robb/.ansible/tmp/ansible-tmp-1461073790.5-184778957863254/" > /dev/null 2>&1'
An exception occurred during task execution. The full traceback is:
Traceback (most recent call last):
  File "/home/robb/.ansible/tmp/ansible-tmp-1461073790.5-184778957863254/sns_topic", line 2497, in <module>
    main()
  File "/home/robb/.ansible/tmp/ansible-tmp-1461073790.5-184778957863254/sns_topic", line 286, in main
    subscriptions]
TypeError: 'NoneType' object is not iterable

fatal: [localhost]: FAILED! => {"changed": false, "failed": true, "invocation": {"module_name": "sns_topic"}, "parsed": false}
```

**Comment by robbwagoner** [comment]
I am not able to reproduce this with Ansible 2.1.0. I see this issue was fixed by @wimnat in ce9aed9c. This can be closed.

**Comment by nitzmahone** [comment]
closing per request
